sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -189,12 +189,13 @@ case class RelationConversions(
   private def convert(relation: HiveTableRelation): LogicalRelation = {
     val serde = relation.tableMeta.storage.serde.getOrElse("").toLowerCase(Locale.ROOT)
     if (serde.contains("parquet")) {
-      val options = relation.tableMeta.storage.properties + (ParquetOptions.MERGE_SCHEMA ->
-        conf.getConf(HiveUtils.CONVERT_METASTORE_PARQUET_WITH_SCHEMA_MERGING).toString)
+      val options = relation.tableMeta.properties ++ relation.tableMeta.storage.properties +
+        (ParquetOptions.MERGE_SCHEMA ->
+          conf.getConf(HiveUtils.CONVERT_METASTORE_PARQUET_WITH_SCHEMA_MERGING).toString)
       sessionCatalog.metastoreCatalog
         .convertToLogicalRelation(relation, options, classOf[ParquetFileFormat], "parquet")
     } else {
-      val options = relation.tableMeta.storage.properties
+      val options = relation.tableMeta.properties ++ relation.tableMeta.storage.properties
```

Comment:
@cloud-fan and @gatorsmile .
Unlike `USING` syntax, `STORED AS` syntax saves the property into `tableMeta.properties`.
This PR considers both table properties.

Reply:
These are not the right things to do. The table properties should not be always put to the serde properties.

Reply:
That is the reason why my previous PR https://github.com/apache/spark/pull/20120 was closed. We need a comprehensive fix for resolving all the DDL issues.

Reply:
This place is only for `convertMetastore`. I think we need this inevitably to prevent regression.

Reply:
It's surprising that Apache Spark doesn't respect table properties in `convertMetastoreParquet` until now. It has been `true` by default.

Reply:
The issues have been defined in the PR description of https://github.com/apache/spark/pull/20120:
```
Currently, we ignore table-specific compression conf when the Hive serde tables are converted to the data source tables. We also ignore it when users set compression in the TBLPROPERTIES clause instead of the OPTIONS clause, even if the tables are native data source tables. 
```
https://github.com/apache/spark/pull/20087 is also trying to resolve the related issues. Thus, we might still miss multiple critical bugs. 

Simply adding the table properties to the options will introduce new bugs. For example, users might provide conflicting serde properties and change the properties through DDLs.

Reply:
We can't blindly add the table properties to the options. This also introduces the behavior changes. For example, some table properties might have the identical names as the serde properties with different semantics. 

Thus, what we can do is to do this only for the table properties we need.

Conceptually, the table properties should not take the serde-related confs. However, Hive basically does not follow the rule it sets. It is pretty strange. No idea why Hive did it.